x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.test.ts

```diff
@@ -0,0 +1,399 @@
+        success: true,
+        success_count: 2,
+      });
+      expect(statusCode).toEqual(200);
+    });
+
+    test('returns 200 with reported conflict if error parsing rule', async () => {
+      const multipartPayload =
+        '--frank_is_awesome\r\n' +
+        `Content-Disposition: form-data; name="file"; filename="rules.ndjson"\r\n` +
+        'Content-Type: application/octet-stream\r\n' +
+        '\r\n' +
+        '{"name"::"Simple Rule Query","description":"Simple Rule Query","risk_score":1,"rule_id":"rule-1","severity":"high","type":"query","query":"user.name: root or user.name: admin"}\r\n' +
+        '{"name":"Simple Rule Query","description":"Simple Rule Query","risk_score":1,"rule_id":"rule-2","severity":"high","type":"query","query":"user.name: root or user.name: admin"}\r\n' +
+        '--frank_is_awesome--\r\n';
+
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+
+      const requestPayload = Buffer.from(multipartPayload);
+      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
+      const parsed: ImportSuccessError = JSON.parse(payload);
+
+      expect(parsed).toEqual({
+        errors: [
+          {
+            error: {
+              message: 'Unexpected token : in JSON at position 8',
+              status_code: 400,
+            },
+            rule_id: '(unknown)',
+          },
+        ],
+        success: false,
+        success_count: 1,
+      });
+      expect(statusCode).toEqual(200);
+    });
+
+    describe('rules with matching rule_id', () => {
+      test('returns with reported conflict if `overwrite` is set to `false`', async () => {
+        clients.alertsClient.find.mockResolvedValue(getFindResult());
+        clients.alertsClient.get.mockResolvedValue(getResult());
+
+        const requestPayload = getSimpleRuleAsMultipartContent(['rule-1', 'rule-1']);
+        const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
+        const parsed: ImportSuccessError = JSON.parse(payload);
+
+        expect(parsed).toEqual({
+          errors: [
+            {
+              error: {
+                message: 'More than one rule with rule-id: "rule-1" found',
+                status_code: 400,
+              },
+              rule_id: 'rule-1',
+            },
+          ],
+          success: false,
+          success_count: 1,
+        });
+        expect(statusCode).toEqual(200);
+      });
+
+      test('returns with NO reported conflict if `overwrite` is set to `true`', async () => {
+        clients.alertsClient.find.mockResolvedValue(getFindResult());
+        clients.alertsClient.get.mockResolvedValue(getResult());
+
+        const requestPayload = getSimpleRuleAsMultipartContent(['rule-1', 'rule-1']);
+        const { statusCode, payload } = await server.inject(
+          getImportRulesRequestOverwriteTrue(requestPayload)
+        );
+        const parsed: ImportSuccessError = JSON.parse(payload);
```

Comment:
👍 Nice typing on the JSON.parse()

Reply:
Thanks! Caught that looking through your e2e tests and added it here.